timeoff/api/viewsets.py

```diff
@@ -0,0 +1,38 @@
+from timeoff.models import Timeoff,Status
+from .serializers import TimeoffSerializer,StatusSerializer
+from rest_framework import viewsets
+from rest_framework.response import Response
+from accounts.models import User
+from django.shortcuts import render, get_object_or_404
+from django.http import HttpResponseRedirect, HttpResponse
+
+
+class TimeoffViewSet(viewsets.ViewSet):
+    # Create, Retreive, Delete, Return list
+    def create(self, request):
+        queryset = Timeoff.objects.all()
+        serializer = TimeoffSerializer
+        if User:
+            serializer.save(username=self.request.username)
+        else:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+    def retrieve(self, request):
+        queryset = Timeoff.objects.all()

```

Comment:
This should be changed to retrieve by user and retrieve should be for getting an individual time off requests by id